wnfs-hamt/src/hamt.rs

```diff
@@ -223,17 +223,41 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
-    use wnfs_common::{dagcbor, MemoryBlockStore};
+    use libipld::cbor::DagCborCodec;
+    use wnfs_common::{async_encode, decode, MemoryBlockStore};
 
     #[async_std::test]
     async fn hamt_can_encode_decode_as_cbor() {
         let store = &MemoryBlockStore::default();
         let root = Rc::new(Node::default());
         let hamt: Hamt<String, i32> = Hamt::with_root(root);
 
-        let encoded_hamt = dagcbor::async_encode(&hamt, store).await.unwrap();
-        let decoded_hamt = dagcbor::decode::<Hamt<String, i32>>(encoded_hamt.as_ref()).unwrap();
+        let encoded_hamt = async_encode(&hamt, store, DagCborCodec).await.unwrap();
+        let decoded_hamt: Hamt<String, i32> = decode(encoded_hamt.as_ref(), DagCborCodec).unwrap();
 
         assert_eq!(hamt, decoded_hamt);
     }
 }
+
+#[cfg(test)]
+mod snapshot_tests {
+    use super::*;
+    use wnfs_common::utils::MockStore;
+
+    #[async_std::test]
+    async fn test_hamt() {
+        let store = &MockStore::default();
+        let node = &mut Rc::new(Node::<[u8; 4], String>::default());
```

Comment:
Nit: Can we simply do `Node::<u32, String>` here?
The `[u8; 4]` makes the snapshot read ... just slightly more weird and confusing.

Reply:
Ah. Right. I regret having the `AsRef<[u8]>` requirement on node's `K` :/
It made it impossible to simply use `NameAccumulator` for `K`, without adding a `serialized_cache: OnceCell<[u8; 256]>` to it.
We should probably add a small trait `HamtKey` or `Hashable` or something like that. We can then additionally have a `impl<T: AsRef<[u8]>> Hashable for T` if we want to, but it'll be possible to implement `Hashable` for `u32` and `NameAccumulator` (without a `serialized_cache`).